packages/parse5/lib/tokenizer/preprocessor.ts

```diff
@@ -17,7 +17,8 @@ const DEFAULT_BUFFER_WATERLINE = 1 << 16;
 export class Preprocessor {
     public html = '';
     private pos = -1;
-    private lastGapPos = -1;
+    // NOTE: Initial `lastGapPos` is -2, to ensure `col` on initialisation is 0
+    private lastGapPos = -2;
```

Comment:
`-2` seems weird, `undefined` or so perhaps? Or would that make the types too complex?

Reply:
That would complicate things quite a bit. The issue here is the check `Number(this.lastGapPos !== this.pos)` below. `-2` is weird, but does make things work as expected.

Reply:
`Infinity` (or `-Infinity`) perhaps?

Reply:
`-Infinity` seems even weirder to me tbh 😄 

`-2` still has a resemblance to the _not found_ return value of `indexOf`, with `-Infinity` I am taken aback by the sudden floating point number for something that is supposed to be an index.